server/service/orbit.go

```diff
@@ -349,6 +352,11 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 			updateChannels = &uc
 		}
 
+		// only unset this flag once we know there were no errors so this notification will be picked up by the agent
+		if notifs.RunDiskEncryptionEscrow {
+			_ = svc.ds.ClearPendingEscrow(ctx, host.ID)
+		}
```

Comment:
What happens if the response never gets to the agent? Given this is an edge case, the solution would be for the user to click `Create key` again, right?

Reply:
That's correct. Per Slack discussion with @mostlikelee / @rachaelshaw, we decided to err on the side of that being the behavior, vs. resending the flag on every config pull until a key is escrowed.